UPnP startup honors user settings

Rallymen007 opened the pull request:
During some research I noticed Kodi would not respect the user's UPnP setting on initial launch.
http://forum.kodi.tv/showthread.php?tid=280683

This simple fix adds a routine setting check before initializing each UPnP subset on startup.

FernetMenta committed:
VideoPlayer: continue normal playback if seek fails

FernetMenta committed:
VideoPlayer: update selectionstreams when stream parameters changes

Paxxi committed:
Change archive to use a fixed length type for string and array lengths to avoid crazy memory allocations.
Also add some rudimentary boundary checking to avoid crashing on corrupt archives

Paxxi committed:
some cleanup of CArchive while touching it

tamland committed:
CFileItemList: add rvalue overload for 'Add'

tamland committed:
CFileItemList: pass shared_ptr by value as it will be copied anyway

tamland committed:
AnnouncementManager: make item parameter const. explicitly copied

tamland committed:
[slideshow] cleanup/cosmetics

tamland committed:
[slideshow] make slideshow window a dialog

tamland committed:
[slideshow] remove transition delay on manual next/prev

feels sluggish

neo1973 committed:
[guilib] Fix memory leak in CGUIFontCacheImpl

neo1973 committed:
[input] Fix memory leak of coding table

neo1973 committed:
[windows] Fixed memory leak in CGUIWindowSplash

tamland committed:
[slideshow] fix access of m_displayEffect before initialization

phate89 committed:
Add the ability to store more than one onlineid

fetzerch committed:
[cmake] Increase link performance (GOLD, DebugFission)

Increase the link performance with GCC by
- defaulting to GNU Gold Linker if available
- defaulting to Debug Fission if available (https://gcc.gnu.org/wiki/DebugFission)

Some measurements on Ubuntu Xenial (GNU GCC 5.3.1, Binutils 2.26)
with incremental linking (touch xbmc/FileItem.cpp) of a debug build:
- GNU ld:   16.5s
- GNU gold:  6.5s

hudokkow committed:
[cmake] Add functions required for CPack DEB

hudokkow committed:
[cmake] Refactor linux Install.cmake by components

hudokkow committed:
[cpack] Add CPackConfigDEB.cmake with specific DEB variables and logic

popcornmix committed:
mmalcodec: Use lock Leave rather than SingleExit

tamland committed:
[addons] add <assets> element for setting icon/fanart location

tamland committed:
[slideshow] add shortcut for starting slideshow from existing list

tamland committed:
[addons] add screenshot support

tamland committed:
[estouchy] add screenshots

popcornmix committed:
mmalcodec: Clear flags before returning picture

hudokkow committed:
[cmake] Nuke legacy XBMCConfig.cmake

popcornmix committed:
mmaldec: Enable zero copy on decoder input buffers

afedchin committed:
[VideoSyncD3D] CVideoSyncD3D should implement IDispResource interface instead of ID3DResource.

koying committed:
FIX: HEADSET_PLUG broadcast intent is code-only

Conflicts:
	xbmc/android/activity/XBMCApp.cpp

neo1973 committed:
[upnp] Fixed the creation of the NPT_LogHandler

The order of arguments to the NPT_LogHandler::Create method was wrong.
Because of that the m_LogHandler variable was set to null. This lead to
a null pointer derefernce in the following line (which for unknown reasons
didn't crash the program).

neo1973 committed:
[fix] Fixed possible division by zero

neo1973 committed:
[thread] Replaced volatile bool with std::atomic<bool> for CThread::m_bStop

volatile does not make a variable thread save, but std::atomic does. This
change made other changes necessary, hence the many changed files.

fetzerch committed:
[cmake] Add CMakeHelpers (utilities for debugging CMake scripts)

fetzerch committed:
[cmake] Opt out if platform configuration cannot be found

For example on Darwin platforms this can indicate a missing toolchain
file.

fetzerch committed:
[cmake] Print CMake version and add some warnings

CMake 3.5.1 suffers from a crash at configure time that happens on some
Systems: http://cmake.org/Bug/view.php?id=16044
This is fixed in 3.5.2.

Darwin requires CMake at least 3.4 or the usage of the patched version
in depends.

hudokkow committed:
[cpack] Add Debian packages metadata

Rallymen007 committed:
UPnP startup honors user settings

MartijnKaijser commented:
@montellese ping

ksooo committed:
[json-rpc][PVR] Expose all CPVRTimerInfoTag attributes to json-RPC.

afedchin committed:
[RenderSystemDX] Hook into user mode driver for selected adapter to fix refresh rate issue in FSE mode.

Montellese commented:
Our coding style is two spaces indentation and curly braces on a separate line. Apart from that the fix is correct. Thanks for bringing it up.

Montellese commented on xbmc/network/NetworkServices.cpp:
```diff
@@ -826,10 +826,16 @@ bool CNetworkServices::StartUPnP()
-  ret |= StartUPnPController();
-  ret |= StartUPnPRenderer();
+  if (CSettings::GetInstance().GetBool(CSettings::SETTING_SERVICES_UPNPSERVER)) {
+	  ret |= StartUPnPClient();
```
The client should always be started because it is required for browsing UPnP sources independent of whether the user wants to share its library or not.

fetzerch committed:
[ColorManager] Replace non-standard VLA with vector

Variable-length arrays are not part of C++ and only allowed in C99. While
they are supported as a GCC extension (if -pedantic is not set), they
cannot be used with MSVC.

This commit replaces the VLA in ColorManager by a std::vector.

neo1973 committed:
[fix] Make CEventGroup::m_bRunning an std::atomic<bool>

to make sure no data race can happen.

neo1973 committed:
[fix] Added missing synchronization to CEventGroup::wait()

ksooo committed:
[json-rpc][PVR] (Add|Delete|Toggle)Timer: Fix implementation and rename 'repeating timer' -> 'timer rule'

ksooo committed:
[json-rpc][PVR] Expose all CEpgInfoTag attributes to json-RPC.

ksooo committed:
[json-rpc][PVR] Expose all CPVRRecording attributes to json-RPC.

ksooo committed:
[json-rpc][PVR] Expose all CPVRChannel attributes to json-RPC.

Commit:
Fix theme xbt loading that broke during cleanup.
Found a race condition that seems to have been hidden and a screwup during cleanup clobbering state.

ksooo committed:
[json-rpc] Add types.json and version.txt to Xcode project.

ksooo committed:
[json-rpc][breaking change] Version bump.

ksooo committed:
[json-rpc][PVR] Expose CPVRTimerInfoTag 'readonly' attribute to json-RPC.

FernetMenta committed:
vdpau: refcount decoder with surfaces given to ffmpeg, align with vaapi, fixes pixelation after seek

popcornmix committed:
VideoPlayer: Remove usused variables

FernetMenta committed:
VideoPlayer: check for duplicate filenames when opening multi source demuxer

FernetMenta committed:
player processinfo: set higher precision for FPS and DAR

fetzerch committed:
[cmake] Fix export-files when building in-source

In-source builds were broken due to a mistake in the export-files
logic. Even though this commit resolves the problem, it is recommended
building out-of-source.

fetzerch committed:
Merge pull request #10017 from fetzerch/cmake-fix-rtmp

[android/packaging] Sync rtmp removal

FernetMenta committed:
Merge pull request #10076 from FernetMenta/multi

VideoPlayer: check for duplicate filenames when opening multi source …

fetzerch committed:
Revert "[android/packaging] Sync rtmp removal"

FernetMenta committed:
guilib: do not kill plugin script if we are waiting for user input

MaxKellermann committed:
MusicDatabase: use range-based "for"

MaxKellermann committed:
MusicDatabase: use "auto"

MaxKellermann committed:
VideoDatabase: use range-based "for"

MaxKellermann committed:
VideoDatabase: use "auto"

MaxKellermann committed:
DbUrl: use "override"

MaxKellermann committed:
DbUrl: make the base class "private"

Nobody ever casts to the base class.  By making it "private", we
narrow the interface, reducing it to a few explicitly mentioned APIs.
This will allow further optimizations, such as making more methods
non-virtual.

MaxKellermann committed:
utils/UrlOptions: remove explicit std::string initialization to empty

This is implied, and it's cheaper.

MaxKellermann committed:
URL: inline trivial methods

These are so simple that a function call is bigger than its inline.

Not inlining the destructor because there may be a bloat of implicitly
generated code in there.

MaxKellermann committed:
utils/UrlOptions: make several methods non-virtual

There are no overrides.

MaxKellermann committed:
utils/UrlOptions: use range-based "for"

MaxKellermann committed:
utils/UrlOptions: include cleanup

MaxKellermann committed:
utils/UrlOptions: use std::string::clear()

This is cheaper than assigning an empty string.

MaxKellermann committed:
utils/Variant: include cleanup

MaxKellermann committed:
utils/Variant: use std::vector::emplace_back()

MaxKellermann committed:
DbUrl: include cleanup

MaxKellermann committed:
utils/UrlOptions: use "auto"

MaxKellermann committed:
test/TestBasicEnvironment: include cores/DataCacheCore.h

Needed because gcc 4.8 complains about the use of std::unique_ptr<>
with a forward-declared type:

 /home/travis/build/xbmc/xbmc/xbmc/ServiceManager.h:47:7:   required from here
 /usr/include/c++/4.8/bits/unique_ptr.h:65:22: error: invalid application of ‘sizeof’ to incomplete type ‘CDataCacheCore’

Newer gcc versions do not have this problem, but unfortunately, the
Kodi build server runs this old version.

afedchin committed:
[dependencies] Added EasyHook to package list.

afedchin committed:
Merge pull request #10082 from afedchin/d3d_vsync_fix

[VideoSyncD3D] CVideoSyncD3D should implement IDispResource interface…

Fneufneu committed:
[cmake/freebsd] build Sinks/AESinkOSS

Fneufneu committed:
[cmake/freebsd] remove ldt_keeper from build

like it was done in Makefile before

Fneufneu committed:
[cmake] fix ffmpeg freebsd build

reuse linux args

Fneufneu committed:
[cmake] do not hardcode make command

use CMAKE_MAKE_PROGRAM variable to fix FreeBSD build

Fneufneu committed:
[cmake] also try to find swig3.0

On FreeBSD the binary are: swig2.0 or swig3.0
so try to find version 3.0 if the 2.0 is not there

stefansaraev committed:
[depends] build curl without librtmp support

kodi now uses ffmpeg's own rtmp implementation, and optionaly inputstream.rtmp
rtmp support was picked automaticlay by libcurl configure if librtmp header/libs
are present in sysroot. no point linking our libcurl against librtmp now

jenkins4kodi committed:
Merge pull request #10078 from fetzerch/cmake_fixes_warnings

FernetMenta committed:
Merge pull request #10075 from FernetMenta/vdpau

vdpau: refcount decoder with surfaces given to ffmpeg, align with vaa…

FernetMenta committed:
Merge pull request #10080 from FernetMenta/plugin

guilib: do not kill plugin script if we are waiting for user input

fetzerch committed:
Merge pull request #10079 from xbmc/revert-10017-cmake-fix-rtmp

Revert "[android/packaging] Sync rtmp removal"

fetzerch committed:
Merge pull request #9924 from fetzerch/cmake_gold

[cmake/linux] Increase link performance (GOLD, DebugFission)

fetzerch committed:
[cmake/cpluff] Add generated artefacts to .gitignore

With CMake, cpluff is still built in the source directory.
- Ignore dwarf objects files (130d154afdc13101f7265356086beedbdaeb8e7f)
- Ignore cpluff-console and cpluff-loader

Fneufneu committed:
[cmake/freebsd] Update freebsd treedata

It's the linux file + xbmc/freebsd

Fneufneu committed:
[cmake/freebsd] Use linux Install.cmake

FernetMenta committed:
Merge pull request #10077 from FernetMenta/codecinfo

player processinfo: set higher precision for FPS and DAR

popcornmix committed:
Merge pull request #10084 from popcornmix/vpunused

VideoPlayer: Remove usused variables

jenkins4kodi committed:
Merge pull request #10073 from Paxxi/fix_theme

jenkins4kodi committed:
Merge pull request #10085 from popcornmix/inzerocopy

tamland committed:
PluginDirectory: dont time out while script is modal/not cancelled

fixes regression from 95ffdcc381288c

tamland committed:
Merge pull request #10059 from tamland/repository_refactor

[addons] repository code cleanup

fetzerch committed:
Merge pull request #10089 from fetzerch/cmake_cpluff

[cmake/cpluff] Add generated artefacts to .gitignore

FernetMenta committed:
fix test suite by declaring a deleter for CDataCacheCore

FernetMenta committed:
Revert "vdpau: refcount decoder with surfaces given to ffmpeg, align with vaa…"

FernetMenta committed:
Merge pull request #10092 from xbmc/revert-10075-vdpau

Revert "vdpau: refcount decoder with surfaces given to ffmpeg, align with vaa…"

fetzerch committed:
[cmake/linux] Code Coverage

Adds a FindCoverage.cmake script that generates code coverages reports
using LCOV. As coverage requires to compile with different options,
a 'Coverage' build type has been introduced.

Usage:
    cmake -DCMAKE_BUILD_TYPE=Coverage <KODI_SRC>/project/cmake \
          && make -j$(nproc) coverage

Known issues: Doesn't work with wrapped libraries (i.g. libcpluff).
Therefore these are compiled unwrapped and the output behaves
differently compared to Debug or Release build!

fetzerch committed:
[cmake] Update README.md

fetzerch committed:
[cmake/android] Install librtmp

After the removal of RTMP in https://github.com/xbmc/xbmc/pull/10017,
the library fails to get packaged for android because it is no more
populated in the dyload-libs variables.

As the library is still needed (https://github.com/xbmc/xbmc/pull/10079)
we manually add it to the files that we're going to package.

Fixes:
cp: cannot stat 'kodi-build-android/install/lib/librtmp.so': No such file or directory
Makefile:121: recipe for target 'libs' failed

fetzerch committed:
Merge pull request #10094 from fetzerch/cmake_fix_android_packaging

[cmake/android] Install librtmp

FernetMenta committed:
Merge pull request #10090 from FernetMenta/testsuite

fix test suite by declaring a deleter for CDataCacheCore

fetzerch committed:
[cddb] Drop non UTF-8 character

BigNoid committed:
[estuary] Add screenshots in addoninfo

fetzerch committed:
[cmake/windows] Fix compilation of tests

fetzerch committed:
[cmake] Add project wide dependency to libcpluff and ffmpeg

The two dependencies libcpluff and ffmpeg are widely required
throughout the whole project. Adding a project wide dependency ensures
that they are built before all libraries added with core_add_library.
This removes the need to specify them per directory which was error
prone in the past (and up to now only fully worked by chance).

fetzerch committed:
Merge pull request #10087 from Fneufneu/freebsd_cmake

Freebsd cmake

wsnipex committed:
Merge pull request #10086 from stefansaraev/curl-nortmp

[depends] build curl without librtmp support

FernetMenta committed:
Merge pull request #10097 from FernetMenta/streams

VideoPlayer: update selectionstreams when stream parameters changes

drieschel committed:
Duplicate include removed

fetzerch committed:
[depends/cmakebuildsys] Pass Configuration to the build

drieschel committed:
Removed duplicate switched to right one

Commit:
Remove cached archives at boot

fetzerch committed:
[buildsteps/linux] Run tests with Code Coverage report generation

Paxxi committed:
Merge pull request #10099 from drieschel/amlutils_duplicate_include

Duplicate include removed

tmelin committed:
Subtitles/SAMI: fix parsing of files with leading whitespace

SAMI (.smi) subtitle files with indentation in lines were not read correctly.
In case subtitle file has indentation for <> tags, strnicmp() failed.

Trim lines to fix comparing of strings.

Function refactored to use helpers from StringUtils.

Signed-off-by: Tomas Melin <tomas.melin@iki.fi>

tamland committed:
[slideshow] reset picture alpha on close

fixes flicker on slideshow start

Commit:
Cleanup work that started with running as Win10 app

UWP(project centennial) does not allow AddDllDirectory which
we don't really need anyway but it's use made the folder structure
pretty. Since we run perfectly fine without it both native and
as UWP I decided to drop it completely.

To make the build, installer and debugging experience easier this
led to plenty of changes to packages and build scripts to output
all dlls into /system/ instead of different folders.

While moving stuff around some dependencies got updated, it's possible
that we can use java menus in bluray movies now, not tested.

EasyHook just got repackaged to fit the new folder structure.

The rebuilding of dependencies led to us being able to drop old
vcredist files which also simplifies installer and packaging, especially
for UWP.

Some minor cleanup along the way as well

fetzerch committed:
[cmake/linux] Make Code Coverage compatible with lcov < 1.10.

This change can be reverted once there is no need to support
lcov < 1.10 anymore (i.g. on Ubuntu 12.04).

ksooo committed:
[PVR] Recordings window: make 'group items' setting persistent.

tamland committed:
[guilib] remove dead code

ksooo committed:
Merge pull request #10101 from ksooo/pvr-group-recordings-setting

[PVR] Recordings window: make 'group items' setting persistent.

garbear committed:
[cmake] Sync Peripheral API

Credit fetzerch, https://github.com/garbear/xbmc/pull/53

FernetMenta committed:
IPlayer: properly sync playspeed with application

Tolriq committed:
Add ipv6 support to Event Server
More than inspired by http://trac.kodi.tv/ticket/12491 with some fixes

FernetMenta committed:
player: drop unneded method ControlsVolume

FernetMenta committed:
applicationPlayer: reset caching timers on OpenFile

tamland committed:
Merge pull request #10093 from tamland/fix_plugin_dir

PluginDirectory: dont time out while script is modal/not cancelled

Paxxi committed:
Merge pull request #9225 from Paxxi/archive

Fix out of memory crash in CArchive

afedchin committed:
[OverlayRendererDX] Fix rendering overlays in TAB/SBS modes.

FernetMenta committed:
Merge pull request #10103 from FernetMenta/state

IPlayer: properly sync playspeed with application

FernetMenta committed:
VideoPlayer: DVDMessageQueue, round up instead to the nearest integer

FernetMenta committed:
VideoPlayer: set clock to video pts when seeking while player is paused

afedchin committed:
Merge pull request #10105 from afedchin/dx_overlay_fix_3d

[OverlayRendererDX] Fix rendering overlays in TAB/SBS modes.

Commit:
Update usage of Crc32 after changing methods to static

afedchin committed:
Merge pull request #10083 from afedchin/d3d11_refresh_hack

[D3D11] Hack to fix refresh rate issue in full screen exclusive mode.

Paxxi committed:
Merge pull request #10107 from Paxxi/fix_crc32

Update usage of Crc32 after changing methods to static

Jalle19 committed:
Merge pull request #10100 from tmelin/smi-parse-fix

Subtitles/SAMI: fix parsing of files with leading whitespace

hudokkow committed:
[cmake] Build kodi-wiiremote

FernetMenta committed:
Merge pull request #10104 from Tolriq/event_server_ipv6

Add ipv6 support to Event Server

koying committed:
FIX: [droid] high cpu with amc surface

FernetMenta committed:
Merge pull request #10109 from FernetMenta/seek

VideoPlayer: fix seeking while paused

MartijnKaijser committed:
Merge pull request #10070 from MaxKellermann/opt1

Various micro-optimizations in the URL code

MartijnKaijser committed:
fix typo after 093912e311525de7bf39f316b8cfa20e400febad

jenkins4kodi committed:
Merge pull request #10111 from fritsch/androidfixes

fetzerch committed:
[cmake] Move platform specific targets into separate files

FernetMenta committed:
AE: fix resetting rr to 1.0

hudokkow committed:
[cmake] Add Debian packaging documentation

Commit:
Missed this one

Paxxi committed:
Merge pull request #10113 from Paxxi/fix_test

[fix][tests] Missed this one

MartijnKaijser committed:
[win32] update AppVeyor with ffmpeg 3.1.1

Commit:
Disable the two failing tests for now so that we can enable the test suite on PR

FernetMenta committed:
player: further cleanup of playing state

MartijnKaijser committed:
[win32] Improve installer and add appxmanifest

MartijnKaijser committed:
[fix] revert commit c9884248b729709450087f05dd8449e794253a6 utils/Variant: use std::vector::emplace_back()
Causes crashing in win32.

MartijnKaijser committed:
Merge pull request #10110 from MartijnKaijser/win32_installer

[win32] Improve installer and add appxmanifest

MartijnKaijser committed:
[win32] fix testbuilds

jenkins4kodi committed:
Merge pull request #10116 from MartijnKaijser/revert

MartijnKaijser committed:
Merge pull request #10117 from MartijnKaijser/win32_test

[win32] fix testbuilds

FernetMenta committed:
Merge pull request #10112 from FernetMenta/ratio

AE: fix resetting rr to 1.0

tamland committed:
[listprovider] open context menu on right click

tamland committed:
[listprovider] respect select action setting

MartijnKaijser committed:
Merge pull request #10114 from Paxxi/shhhh

[tests] Disable the two failing tests for now

fetzerch committed:
[cmake] Add binary-addons target for makefile generator

Adds a target that allows to build the binary addons:
make binary-addons ADDONS="visualization.spectrum"

This can be used if local addons should be built for packaging (for
example on osx or android).

popcornmix committed:
rbp: Make debug message wrap safe

MartijnKaijser committed:
[win32] update wget and drop old shipped DLL files

Commit:
Fix utf8 handling during build causing test to fail

MartijnKaijser committed:
Merge pull request #10121 from Paxxi/fix_tests

[fix][tests] Fix utf8 handling during build causing test to fail

razzeee committed:
Move win32 files to their own platform folder

MartijnKaijser committed:
Merge pull request #10122 from MartijnKaijser/win32_wget

[win32] update wget and drop old shipped DLL files

FernetMenta committed:
Merge pull request #10115 from FernetMenta/speed

player: further cleanup of playing state

MartijnKaijser committed:
Merge pull request #10063 from neo1973/fixes

Random fixes found with clang/gcc sanitizers

tamland committed:
Merge pull request #10060 from tamland/addon_screenshots

[addons] screeshot support

FernetMenta committed:
OpenGL: cleanup vsync

tamland committed:
Merge pull request #10118 from tamland/fix_listprovider_rightlick

[listprovider] open context menu on right click

tamland committed:
[addons] fix bad refactoring in fb51bbd26e0

icon should be set even if metadata element is not present

tamland committed:
Merge pull request #10128 from tamland/fix_addon_tests

[addons] fix bad refactoring in fb51bbd26e0

FernetMenta committed:
Merge pull request #10127 from FernetMenta/vsync

OpenGL: cleanup vsync

fetzerch committed:
[cmake] Build unrarxlib and libupnp without core_add_library

fetzerch committed:
[cmake] Change the way shared libraries are built

Before this change shared libraries were built by creating a static
library using core_add_library and then linking that into a
shared/module one using core_link_libraries. This has a few drawbacks:

 - Meanwhile most libraries don't use wrapping (for VFS support)
   anymore, and we can use CMake mechanisms to create those libraries.
 - The approach doesn't work with MSVC as there is no "whole-archive"
   option. On windows these symbols would need different exporting.
 - The main usage of core_add_library is to generate small libraries
   for the Kodi main application. Let's use it only for that.
   This is in preparation to disable building of per folder static
   libraries because they don't work well with VS and Xcode.

This commit introduces the following changes:
 - Shared/Module libraries are now created with core_add_shared_library,
   addon callback libraries are now created with core_add_addon_library.
 - Fix dependencies: make kodi now builds everything needed to run kodi
   including all dl-loaded libraries.
 - Only use wrapping for libraries where it's also done with Autotools
 - WRAP_FILES and wrap-libraries are renamed to LIBRARY_FILES and
   kodi-libraries.
 - Library wrapping and generation now works in all subdirectories.
   Previously core_link_libraries had to be called in the main
   CMakeLists.txt because file dependencies in add_custom_command only
   work in the same directory. This also made it necessary to create
   wrap_* targets.
 - Generator expressions in core_link_libraries make the function
   less error prone.

MartijnKaijser committed:
Merge pull request #10123 from Paxxi/dll_dir

[xbmc][libs] Cleanup work that started with running as Win10 app

Commit:
Forgot to update the path here to reflect the new dll location

tamland committed:
[addons] reinstate multidir repositories. partially reverts eec2e0abacac1a32fd5a42fafc90336237f8bcf7

MartijnKaijser committed:
Merge pull request #10131 from Paxxi/fix_runtest

[test][fix] Fix running of test suite on win32

tamland committed:
[addons] fix duplicate addons in repo list

bad refactoring in 8f7b6a25d42f7

tamland committed:
Merge pull request #10133 from tamland/repo_dupes

[addons] fix duplicate addons in repo list

ksooo committed:
Merge pull request #10074 from ksooo/pvr-timers-json-rpc

[PVR][json-rpc] change/fix timer API, sync pvr types fields with current implementation

Tolriq committed:
Add new option installed to Addons.GetAddons to allow listing of addons from loaded repositories.
Add installed field to Addon.Fields / Addon.Details to indicate installation status.

mk01 committed:
[PeripheralCecAdapter] fix after 6a5120c3789a7ad93a235467a4cb421dee643438

wsnipex committed:
Merge pull request #10095 from fetzerch/cmake_codecoverage

[cmake/linux] Code coverage

jenkins4kodi committed:
Merge pull request #10140 from hudokkow/json_all_addons

tamland committed:
Merge pull request #10135 from tamland/multidir_repo

[addons] reinstate multidir repositories. partially reverts eec2e0aba…

tamland committed:
Merge pull request #10126 from tamland/listprovider_select_action

[listprovider] respect select action setting

fetzerch committed:
[cmake] Let ExternalProject_Add figure out the right 'make' command

ExternalProject.cmake sets a default BUILD_COMMAND to 'make'. Not
overriding it has the advantage that it figures out the best option
itself (including passing down parallel build parameters).

This fixes the Ninja build on Linux due to a problem introduced in
810895469a621c2eb25dbb2ef4aba75ecf067c8b where ninja would be used to
build ffmpeg and cpluff (which doesn't work).

popcornmix committed:
mmalcodec: Remove deinterlace support

phate89 committed:
Fix retrieve all the information before update

phate89 committed:
Merge pull request #10142 from phate89/fix_retrieve

Fix retrieve all the information before update

fetzerch committed:
[cmake/win32] Set startup project and generate launcher for Visual Studio

MartijnKaijser committed:
Merge pull request #10138 from Razzeee/headset

FIX: HEADSET_PLUG broadcast intent is code-only

popcornmix committed:
mmal: Switch to a pool class attached to decoder output

Pool containing mmal buffers (e.g. decoded pictures) need to be attached
to a mmal components port.

Managing the lifespan of these is tricky, as renderer may consitue to
display an image after the decoder has been closed and a new one opened.

So create a class for handling this. The class creates and destroys the
mmal component associated with the pool, and creates the mmal pool
(which is just headers) as well as handling allocation of the gpu buffers.

Each object from the pool includes a shared pointer to the pool class,
ensuring the pool and mmal component are not destroyed until the last
object is destroyed.

The pool class is used by both MMAL hw decode and the FFMPEG zero copy
decoders which avoids some special casing in the renderer.

It will also handle the pool of deinterlaced frames later.

Commit:
Bump sdk to Win 10 as cmake defaults to that for addons anyway. Add the few headers we need from wdk as a package to simplify setup

popcornmix committed:
mmal: Remove renderer pointer from codecs

popcornmix committed:
mmal: Fix some ugliness with acquire/release of video buffers

popcornmix committed:
mmalrender: Execute callbacks from dedicated thread

mmal callbacks all come from a common thread, so blocking in a callback
is inefficient at best and can risk deadlocks.

Creating a dedicated thread for processing mmal buffers
(including the reconfiguration when formats change) avoids this.

popcornmix committed:
mmalrender: Add deinterlace support

popcornmix committed:
mmalrender: Allow a frame of slop when waiting for vsync

Previously we would always wait an extra vsync (even if one had been missed) in RenderUpdate.
Now if a single vsync is missed, we won't wait.

phate89 committed:
Merge pull request #10000 from phate89/multiple_ids

Add the ability to store more than one online id

Paxxi committed:
Merge pull request #10143 from Paxxi/wdk

[depends][xbmc][fix] Get rid of wdk requirement

popcornmix committed:
Merge pull request #10144 from popcornmix/vsyncslop

mmalrender: Allow a frame of slop when waiting for vsync

hudokkow committed:
Merge pull request #9987 from hudokkow/cpack_debian

[cmake/cpack] Debian packages generation

BigNoid committed:
[estuary] Show fallback behind thumb in addoninfo preview list

popcornmix committed:
Merge pull request #10120 from popcornmix/movedeint

MMAL: Move deinterlacer from decoder to renderer

phate89 committed:
Remove pointless check in python setUniqueIDs (and fix description)

I'm not sure why I added it in the first place.. If there is no
videoinfotag we create one..

FernetMenta committed:
GLX: fix vsync for glXSwapIntervalMESA after f0c02bd1adb8be38d8a5ae56c2f77447d1fc6fee

fetzerch committed:
Merge pull request #10130 from fetzerch/cmake_splitlibs

[cmake] Change the way how shared libraries are built

txtranslation committed:
[lang][skin.estuary] updated language files from Transifex

txtranslation committed:
[lang][skin.estouchy] updated language files from Transifex

MartijnKaijser committed:
[win32] Don't add version to the name as that will also change menu shortcut creation. Just put it in description.

MartijnKaijser committed:
[win32] add easyhook.dll to gitignore

MartijnKaijser committed:
Merge pull request #10152 from MartijnKaijser/win32_fix

Fix win32 installer and add DLL to gitignore

FernetMenta committed:
Merge pull request #10150 from FernetMenta/vsync

GLX: fix vsync for glXSwapIntervalMESA after f0c02bd1adb8be38d8a5ae56…

BigNoid committed:
Merge pull request #10146 from BigNoid/estuary_addoninfo

[estuary] Show fallback behind thumb in addoninfo preview list

koying committed:
FIX: [droid] feed codecinfo

koying committed:
FIX: [droid;videosync] OnResetDevice was renamed to OnResetDisplay

koying committed:
FIX: [amc] implement DVD_CODEC_CTRL_DRAIN

phate89 committed:
Merge pull request #10148 from phate89/remove_python_uid_check

Remove pointless check in python setUniqueIDs (and fix description)

tamland committed:
CColorManager::Create3dLut: fix return-type

koying committed:
FIX: [amc] freeze when refresh rate changed

jenkins4kodi committed:
Merge pull request #10154 from tamland/fix_return_type

koying committed:
FIX: [amc] tweaks

fritsch committed:
AESinkPULSE: Do not remix channels by default

Commit:
Fix on demand dll loading to point to the correct files

MartijnKaijser committed:
Merge pull request #10155 from Paxxi/dll_loader

[xbmc][fix] Fix on demand dll loading to point to the correct files

fritsch committed:
AESinkPULSE: Require at least version 2.0 of PA

fritsch committed:
AESinkPULSE: Tag passthrough content

fetzerch committed:
[cmake/linux] Move WiiRemote target to ExtraTargets.cmake

MartijnKaijser committed:
Merge pull request #10153 from Razzeee/feed-info

Codecinfo, Drain and ResetDisplay fixes

hudokkow committed:
Merge pull request #10159 from fetzerch/cmake_wii

[cmake/linux] Move WiiRemote target to ExtraTargets.cmake

fritsch committed:
AESinkAudiotrack: Do not average over 4 seconds but 200 ms

FernetMenta committed:
Merge pull request #10139 from mk01/master_ceconscreensaver

[PeripheralCecAdapter] fix broken standby after …

MartijnKaijser committed:
Merge pull request #10124 from Razzeee/move-win32-platform

Move win32 files to their own platform folder

jenkins4kodi committed:
Merge pull request #10161 from fritsch/androidsync

fritsch committed:
Merge pull request #10151 from fritsch/silenceforpa

AESinkPULSE: Disable remixing if no server downmix is needed

Rallymen007 committed:
Merge branch 'master' of https://github.com/Rallymen007/xbmc

Rallymen007 committed:
UPnP client always starts, fixed code style

Rallymen007 commented:
The rebase was half a good idea as it turns out, but I fixed the indentation (my mistake, VS added tabs and I wasn't paying attention) and moved the UPnP client initialization.

In my use case I wanted to completely turn off the UPnP portion as it seemed to interfere with my custom UPnP client, hence why StartClient was there. I ended up just recompiling Kodi with -disable_upnp and it was all good.

MartijnKaijser commented:
@Rallymen007 sadly the rebase went wrong. Best it to always make a branch of our master and name it different. A quick guide can be found here
https://github.com/xbmc/xbmc/blob/master/CONTRIBUTING.md

Could you create a new pull rquest please in which you do all changes in a single commit?
